verification/certifyKey_test.go

```diff
@@ -121,18 +404,221 @@ func testCertifyKey(d TestDPEInstance, t *testing.T) {
 		t.Fatalf("Could not initialize client: %v", err)
 	}
 
-	certifyKeyReq := CertifyKeyReq[SHA256Digest]{
-		ContextHandle: [16]byte{0},
-		Flags:         0,
-		Label:         [32]byte{0},
-		Format:        CertifyKeyX509,
+	certifyKeyReq := []CertifyKeyReq[SHA256Digest]{
+		{
+			ContextHandle: [16]byte{0},
+			Flags:         0,
+			Label:         [32]byte{},
+			Format:        CertifyKeyX509,
+		},
+		{
+			ContextHandle: [16]byte{0},
+			Flags:         1073741824, // Set the 31st bit: 01000000 00000000 00000000 00000000
+			Label:         [32]byte{},
+			Format:        CertifyKeyX509,
+		},
+	}
+
+	for _, r := range certifyKeyReq {
+		// Get DPE leaf certificate from CertifyKey
+		certifyKeyResp, err := client.CertifyKey(&r)
+		if err != nil {
+			t.Fatalf("Could not certify key: %v", err)
+		}
+
+		// Get root and intermediate certificates to validate certificate chain of leaf cert
+		getCertificateChainResp, err := client.GetCertificateChain()
+		if err != nil {
+			t.Fatalf("Could not get Certificate Chain: %v", err)
+		}
+
+		leafCertBytes := certifyKeyResp.Certificate
+		certChainBytes := getCertificateChainResp.CertificateChain
+
+		// Run X.509 linter on full certificate chain and file issues for errors
+		leafCert := checkCertificateStructure(t, leafCertBytes)
+		t.Logf("Leaf certificate is DER encoded")
+
+		certChain := checkCertificateChain(t, certChainBytes)
+		t.Logf("Certificate chain is DER encoded")
+
+		// Validate that all X.509 fields conform with the format defined in the DPE iRoT profile
+		validateCertifyKeyCert(t, leafCert, uint32(r.Flags), r.Label[:])
+
+		// Ensure full certificate chain has valid signatures
+		validateCertChain(t, certChain, leafCert)
+
+		// TODO: When DeriveChild is implemented, call it here to add more TCIs and call CertifyKey again.
 	}
+}
 
-	certifyKeyResp, err := client.CertifyKey(&certifyKeyReq)
+// Validate signature of certificates in certificate chain passed.
+func validateSignature(t *testing.T, certchain []*x509.Certificate) {
+	t.Helper()
+	foundError := false
+	for i := 0; i < len(certchain)-1; i++ {
+		err := certchain[i].CheckSignatureFrom(certchain[i+1])
+		if err != nil {
+			t.Logf("Signature validation failed for certificate %s in chain with error %s", certchain[i].Subject.String(), err.Error())
+			foundError = true
+		} else {
+			t.Logf("Signature validation passed %s", certchain[i].Subject.String())
+		}
+
+	}
+
+	root := certchain[len(certchain)-1]
+	//err := root.CheckSignatureFrom(root)
```

Comment:
This would just check that the root is self-signed, which is not a requirement. This should be removed.

Reply:
Was removed along with the redundant function,